avm/src/vm/http.rs

```diff
@@ -57,7 +57,7 @@ macro_rules! make_server {
         let port_num = http.port;
         let addr = std::net::SocketAddr::from(([0, 0, 0, 0], port_num));
         let make_svc = hyper::service::make_service_fn(|_conn| async {
-          Ok::<_, std::convert::Infallible>(hyper::service::service_fn($listener))
+          Ok::<_, $crate::vm::VMError>(hyper::service::service_fn($listener))
```

Comment:
Why this change?

Reply:
the service function could return a `VMError` since it has to work with the memory of the event that handles each request, so I allow the hyper service to propagate such errors - although we could handle the error directly in the service function?

Reply:
So for the web server I would prefer it is infallible and if it encounters any `VMError` responses it returns a `500 Internal Server Error` to the end user with some sort of short message indicating an unexpected failure within the AVM itself? Simply crashing out will make it harder to track down what's going wrong, I think.